README.md

```diff
@@ -27,7 +27,53 @@ This project welcomes contributions and suggestions.  Most contributions require
 ## Script support
 PSF allows one PowerShell script to be run before an exe runs, and one PowerShell script to be ran after the exe runs.
 
-Each exe defined in the application manifest can have their own scripts.
+Each exe defined in the application manifest can have their own scripts.
+
+### Stopping application on a script error.
+PSF gives users the options to stop the application if the starting script encounters an error.  To enable PSF to stop if the starting script encounters an error
+add the pair stopOnScriptError and set it to true.  Example `"stopOnScriptError": true`  The default PSF value for stopOnScriptError is false.  This pair can be specified on a per exe basis.
+
+If `stopOnScriptError` is either false or not defined the behavior is to run the application and endingScript (if defined) even if there is an error with the starting script.
+
+#### Determining what constitutes an error for scripts.
+PSF does not alter the Error Action Preference for powershell.  If you need to treat non-terminating errors as terminating errors please change the Error Action Preference in each
+of your powershell scripts.  You can find more information about setting the Error Action Preference here: [MSDN setting the Error Actio Preference.](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.psinvocationsettings.erroractionpreference?view=pscore-6.2.0)
+
+### Hiding the script windows
+PSF allows the user to hide the PowerShell window for both the starting script and ending script.  To do so add `"showWindow": true` to the script object.  The default behavior is to hide the window.
+
+### Running the script once
+PSF allows the starting script and the ending script to be ran once per application version.  To enable this run once feature add `"runOnce" : true` to any script object.  Adding this means that the script will only run once per version.  The default value for this pair is false.
+
+### Timeout
+Both the starting script and ending script support timeouts in the form of seconds.  Timeouts are optional.
+
+If no timeout is supplied PSF will wait for an infinite amount of time for the script to exit.

```

Comment:
will wait indefinitely for the script to exit. The user application will not start until the script runs to completion or is terminated.